### DISCUSSION TITLE
are devices like the max3421e and ch372 (& 376 which does host and some automatic stuff) outside of the scope of this project?

### DISCUSSION THREAD

**Discussion by ohsix** [hathach/tinyusb#1449] [General]
and if not, which driver is the simplest or most straightforward for someone not super familiar with usb to try and implement support for them

thanks!
  josh

**Comment by hathach** [comment]
no, why would it is outside of this project scope though ?

**Comment by ohsix** [comment]
just thought it be better to ask

some aspects of the ch372 implementation happen automatically and i was unsureif every device had to work like native usb ip or if the high level usb uart/spi interfaces can usefully map to it (ch376 is the ch372 "plus", can do host mode w/ full control of the endpoints or MSC/HID/some other stuff by itself)

also i was looking in completely the wrong place to find an example driver, whoops

host/hcd.h & dcd_template.c was what i was looking for

thanks!

**Reply by hathach** [reply]
to be honest, I don't know much of ch372,though there is an PR to add support for another WCH here #1405 . As long as it is software controllable, I would try to support it if there is enough time for the work.
